chia/full_node/block_store.py

```diff
@@ -557,9 +705,26 @@ async def get_block_records_in_range(
                 async with await conn.execute(formatted_str) as cursor:
                     for row in await cursor.fetchall():
                         header_hash = self.maybe_from_hex(row[0])
-                        ret[header_hash] = BlockRecord.from_bytes(row[1])
+                        ret[header_hash] = BlockRecordDB.from_bytes(row[1])
 
-        return ret
+            formatted_str = f"SELECT header_hash, block from full_blocks WHERE height >= {start} and height <= {stop}"
+
+            async with self.db_wrapper.reader_no_transaction() as conn:
+                async with await conn.execute(formatted_str) as cursor:
+                    for row in await cursor.fetchall():
+                        header_hash = self.maybe_from_hex(row[0])
+                        all_blocks_plot_filters[header_hash] = plot_filter_info_from_block(row[1])
+
+        result: Dict[bytes32, BlockRecord] = {}
+        for hh, block_record_db in ret.items():
```

Comment:
having this loop be separate looks a bit unnecessarily complex. I think it would make more sense to compute the final `BlockRecord` object in the loop where we're reading from the DB cursor directly. 
Then you wouldn't need the extra `all_blocks_plot_filters` dict either, and you can just return `ret` at the end